server/cp/ai/semaphore.py

```diff
@@ -70,49 +38,68 @@ class Semaphore(AIServiceBase):
+    def __init__(self, data):
         # SEMAPHORE_BASE_URL OR TOKEN_ENDPOINT Goes Here
-        self.base_url = app.config.get("SEMAPHORE_BASE_URL")
+        self.base_url = os.getenv("SEMAPHORE_BASE_URL")
 
         #  SEMAPHORE_ANALYZE_URL Goes Here
-        self.analyze_url = app.config.get("SEMAPHORE_ANALYZE_URL")
+        self.analyze_url = os.getenv("SEMAPHORE_ANALYZE_URL")
 
         #  SEMAPHORE_API_KEY Goes Here
-        self.api_key = app.config.get("SEMAPHORE_API_KEY")
+        self.api_key = os.getenv("SEMAPHORE_API_KEY")
 
         #  SEMAPHORE_SEARCH_URL Goes Here
-        self.search_url = app.config.get("SEMAPHORE_SEARCH_URL")
+        self.search_url = os.getenv("SEMAPHORE_SEARCH_URL")
 
         #  SEMAPHORE_GET_PARENT_URL Goes Here
-        self.get_parent_url = app.config.get("SEMAPHORE_GET_PARENT_URL")
+        self.get_parent_url = os.getenv("SEMAPHORE_GET_PARENT_URL")
 
         #  SEMAPHORE_CREATE_TAG_URL Goes Here
-        self.create_tag_url = app.config.get("SEMAPHORE_CREATE_TAG_URL")
+        self.create_tag_url = os.getenv("SEMAPHORE_CREATE_TAG_URL")
 
         #  SEMAPHORE_CREATE_TAG_TASK Goes Here
-        self.create_tag_task = app.config.get("SEMAPHORE_CREATE_TAG_TASK")
+        self.create_tag_task = os.getenv("SEMAPHORE_CREATE_TAG_TASK")
 
         #  SEMAPHORE_CREATE_TAG_QUERY Goes Here
-        self.create_tag_query = app.config.get("SEMAPHORE_CREATE_TAG_QUERY")
+        self.create_tag_query = os.getenv("SEMAPHORE_CREATE_TAG_QUERY")
+
+        self.output = self.analyze(data)
+
+    def convert_to_desired_format(input_data):
+        result = {
+            "result": {
+                "tags": {
+                    "subject": input_data["subject"],
+                    "organisation": input_data["organisation"],
+                    "person": input_data["person"],
+                    "event": input_data["event"],
+                    "place": input_data["place"],
+                    "object": [],  # Assuming no data for 'object'
+                },
+                "broader": {"subject": input_data["broader"]},
+            }
+        }
+
+        return result
 
     def get_access_token(self):
         """Get access token for Semaphore."""
         url = self.base_url
-        assert url
 
         payload = f"grant_type=apikey&key={self.api_key}"
         headers = {"Content-Type": "application/x-www-form-urlencoded"}
         response = session.post(url, headers=headers, data=payload, timeout=TIMEOUT)
         response.raise_for_status()
         return response.json().get("access_token")
 
-    def fetch_parent_info(self, qcode):
+    def fetch_parent_info(self, qcode, article_language):
         headers = {"Authorization": f"Bearer {self.get_access_token()}"}
         try:
             frank = "?relationshipType=has%20broader"
 
             query = qcode
             parent_url = self.get_parent_url + query + frank
+            logger.warning("parent_url: %s", parent_url)

```

Comment:
that should be probably rather info/debug level

Reply:
Resolved